doc/api/metrics.md

```diff
@@ -0,0 +1,635 @@
+# Metrics
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/internal/perf/metrics.js -->
+
+The metrics API provides an API for application instrumentation and
+performance monitoring. It offers various metric types and built-in exporters
+for popular monitoring systems.
+
+The metrics API can be accessed using:
+
+```mjs
+import { metrics } from 'node:perf_hooks';
+```
+
+```cjs
+const { metrics } = require('node:perf_hooks');
+```
+
+## Overview
+
+The metrics API enables developers to instrument their applications with custom
+metrics that can be collected and exported to monitoring systems. All metrics
+publish their data through the `node:diagnostics_channel` module, allowing for
+flexible consumption patterns.
+
+### Example
+
+```mjs
+import { metrics } from 'node:perf_hooks';
+const { createCounter } = metrics;
+
+// Create counter metrics
+const apiCalls = createCounter('api.calls', { service: 'web' });
+const requestDuration = createCounter('api.request.duration.ms', { service: 'web' });
```

Comment:
Using counters to record durations is not a good practice in real life. It is not a single counter value been incremented. It should be represented by a distribution of values, like Histogram.

The API should distinguish between counters and histograms.